runtime/interpreter/value.go

```diff
@@ -17834,14 +17837,14 @@ func (i *InclusiveRangeIterator) Next(interpreter *Interpreter) Value {
 	valueToReturn := i.next
 
 	// Ensure that valueToReturn is within the bounds.
-	if i.stepNegative && bool(valueToReturn.Less(interpreter, i.end, EmptyLocationRange)) {
+	if i.stepNegative && bool(valueToReturn.Less(interpreter, i.end, i.locationRange)) {
 		return nil
-	} else if !i.stepNegative && bool(valueToReturn.Greater(interpreter, i.end, EmptyLocationRange)) {
+	} else if !i.stepNegative && bool(valueToReturn.Greater(interpreter, i.end, i.locationRange)) {
 		return nil
 	}
 
 	// Update the next value.
-	nextValueToReturn, ok := valueToReturn.Plus(interpreter, i.step, EmptyLocationRange).(IntegerValue)
+	nextValueToReturn, ok := valueToReturn.Plus(interpreter, i.step, i.locationRange).(IntegerValue)
```

Comment:
For the use-case here, for-looping using the iterator it actually doesn't matter, but the place of the iterator use (call to `Next`) and the place of the iterator construction (`Iterator` function) might be different.

Could you please add a location range parameter to `Next` instead and use that parameter here and in the two cases above, then remove the `locationRange` field from `InclusiveRangeIterator` again? (It looks like it wasn't populated anyway)

Reply:
I believe you're looking at an older commit. Can you please check again? b512835679ec724dadded298f08ce3fc159d3d78 is the latest commit.

Reply:
Ah, right. Sorry, my bad!